src/Controllers/PasswordResetController.php

```diff
@@ -47,22 +48,7 @@ public function postReset(Request $request): Response
         /** @var User $user */
         $user = User::whereEmail($data['email'])->first();
         if ($user) {
-            $reset = (new PasswordReset())->findOrNew($user->id);
-            $reset->user_id = $user->id;
-            $reset->token = bin2hex(random_bytes(16));
-            $reset->save();
-
-            $this->log->info(
-                sprintf('Password recovery for %s (%u)', $user->name, $user->id),
-                ['user' => $user->toJson()]
-            );
-
-            $this->mail->sendViewTranslated(
-                $user,
-                'Password recovery',
-                'emails/password-reset',
-                ['username' => $user->displayName, 'reset' => $reset]
-            );
+            PasswordResetHelper::triggerPasswordReset($user);
```

Comment:
I would have left it in this Controller and just moved it  to a public method which can be called by the settings page; The Helpers are "not a great idea" design wise / by themselves as its a place where "all that random junk" accumulates (generally it would be great to use them only for stuff that actually has no other place to live).

tldr: I am not completely happy but would let it go through?

Reply:
@xuwhite implemented an alternative approach in https://github.com/engelsystem/engelsystem/commit/e257c3d715c37369223b50161e4238370fe3e199 by integrating it to the PasswordResetController -> feel free to integrate it / refactor that a bit (eventually switching the method calls around could be even cleaner) :)